### PR TITLE
Enable nullability in some TreeNode members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some TreeNode members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8126)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs [hunk 1]
```diff
@@ -331,9 +334,9 @@ public bool Checked
             get
             {
 #if DEBUG
-                if (_handle != IntPtr.Zero && !treeView.IsDisposed)
+                if (_handle != IntPtr.Zero && treeView is not null && !treeView.IsDisposed)
                 {
-                    TreeView tv = TreeView;
+                    TreeView tv = TreeView!;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:339]: is `TreeView` and t`reeView` same here?

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs [hunk 2]
```diff
@@ -674,7 +678,7 @@ public TreeNode LastNode
                     return null;
                 }
 
-                return children[childCount - 1];
+                return children![childCount - 1];
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:681]: Why can't we tie `childCount` and `children` together? Can you try to remove `childCount` and let `children` be the source of truth?

**Review by JeremyKuhne — CHANGES_REQUESTED** [review]
It's worth trying to fix up the disconnect between the array and the count as it's really dangerous to have them separate. Additionally please make sure the `Deserialize` doesn't let garbage (nulls, etc.) back in where we would never have serialized garbage in the first place. That will put us in a better position to null annotate.

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:681]: Hmm, `childCount` according to `TreeNodeAndPropertyBag_RoundTripAndExchangeWithNet` participates in serialization/deserialization. So, I assume that we can't just remove it, right? Or you mean something else?

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:681]: I don't think it "participates" in the serialization, it just gets validated after the deserialization is complete to contain the expected value.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:681]: The field is read here, but the value can be equally read from another source:
https://github.com/dotnet/winforms/blob/852e29689aca09c81bba1c73eade41a96e13704b/src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs#L2152

Here we probably don't technically need to assign the field, but use it as a local variable to initialise the underlying value collection:
https://github.com/dotnet/winforms/blob/852e29689aca09c81bba1c73eade41a96e13704b/src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs#L1597-L1598
https://github.com/dotnet/winforms/blob/852e29689aca09c81bba1c73eade41a96e13704b/src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs#L1636-L1646

**Comment by dreddy-work** [comment]
@gpetrou, checking back on this long pending PR. Let us know if you are waiting on us.

**Comment by gpetrou** [comment]
@dreddy-work any feedback on this one?

**Comment by gpetrou** [comment]
@dreddy-work maybe you could also take a look at this one?

**Review by JeremyKuhne — APPROVED** [review]
Just one nit, otherwise looks good.

**Pull request merged**